cocos/core/scene-graph/base-node.ts

```diff
@@ -222,32 +218,28 @@ export class BaseNode extends CCObject implements ISchedulable {
         return this._eventProcessor;
     }
 
-    public static _setScene (node: BaseNode) {
-        if (node instanceof legacyCC.Scene) {
-            node._scene = node;
-        } else {
-            if (node._parent == null) {
-                error('Node %s(%s) has not attached to a scene.', node.name, node.uuid);
-            } else {
-                node._scene = node._parent._scene;
-            }
-        }
-    }
-
     protected static idGenerator = idGenerator;
 
     // For walk
     protected static _stacks: Array<Array<(BaseNode | null)>> = [[]];
     protected static _stackId = 0;
 
-    protected static _findComponent (node: BaseNode, constructor: Function) {
+    /**
+     * Call `_updateScene` of specified node.
+     * @param node The node.
+     */
+    protected static _setScene (node: BaseNode) {
+        node._updateScene();
```

Comment:
I do think this is better than `if (node instanceof ...)`. How do you think? @jareguo

Reply:
Agree